### PR TITLE
[full-ci] [tests-only] Use `sharing-ng` in given steps for sharing

### PR DESCRIPTION
**Pull request by grgprarup**
## Description
Used sharingNG graph API in the given steps for sharing.

Suites covered:
- `coreApiSharePublicLink1/`
  - `accessToPublicLinkShare.feature`
  - `changingPublicLinkShare.feature`
  - `createPublicLinkShare.feature`
  - `deletePublicLinkShare.feature`

## Related Issue

- Part of https://github.com/owncloud/ocis/issues/8717

## How Has This Been Tested?

- Locally
- :robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/bootstrap/PublicWebDavContext.php [hunk 1]
```diff
@@ -332,17 +332,12 @@ public function downloadPublicFileInsideAFolderWithRange(string $path, string $r
 	/**
 	 * @param string $path
 	 * @param string $user
-	 * @param bool $shareNg
 	 *
 	 * @return ResponseInterface
 	 */
-	public function downloadFromPublicLinkAsUser(string $path, string $user, bool $shareNg = false): ResponseInterface {
+	public function downloadFromPublicLinkAsUser(string $path, string $user): ResponseInterface {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/PublicWebDavContext.php:338]: this method might be used somewhere with $shareNg arg
```suggestion
	public function downloadFromPublicLinkAsUser(string $path, string $user): ResponseInterface {
```

File: tests/acceptance/features/coreApiSharePublicLink1/changingPublicLinkShare.feature [hunk 2]
```diff
@@ -171,36 +197,18 @@ Feature: changing a public link share
       | 1               | 200              |
       | 2               | 403              |
 
-
-  Scenario Outline: normal user removes password of a public link (invite only public link)
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiSharePublicLink1/changingPublicLinkShare.feature:175]: is it duplicate scenario?

File: tests/acceptance/features/bootstrap/PublicWebDavContext.php (hunk 1 above)
Comment (grgprarup) [reply tests/acceptance/features/bootstrap/PublicWebDavContext.php:338]: The method is used only once without passing `$shareNg` so it's safe to remove. Also, the condition in L#340 is changed.

File: tests/acceptance/features/coreApiSharePublicLink1/changingPublicLinkShare.feature (hunk 2 above)
Comment (grgprarup) [reply tests/acceptance/features/coreApiSharePublicLink1/changingPublicLinkShare.feature:175]: We cannot add passwords for `internal` public links. Do we need such a test anymore?

Comment (saw-jan) [reply tests/acceptance/features/coreApiSharePublicLink1/changingPublicLinkShare.feature:175]: there should be a scenario in sharingNG suite

**Review by saw-jan — APPROVED** [review]

**Pull request merged**